client/network/src/discovery.rs

```diff
@@ -731,6 +770,10 @@ impl NetworkBehaviour for DiscoveryBehaviour {
 							let ev = DiscoveryOut::Discovered(peer);
 							return Poll::Ready(NetworkBehaviourAction::GenerateEvent(ev))
 						},
+						KademliaEvent::InboundPutRecordRequest { .. } |
+						KademliaEvent::InboundAddProviderRequest { .. } => {
+							unreachable!("We don't use kad filtering at the moment");
```

Comment:
Could you change this for a `debug_assert!(false, "...")` and ignore the event instead? This way we panic in debug mode but the node will continue to run in release.